src/main/java/com/google/devtools/build/lib/skyframe/toolchains/SingleToolchainResolutionFunction.java

```diff
@@ -91,14 +90,21 @@ public SkyValue compute(SkyKey skyKey, Environment env)
                 .debugToolchainResolution(key.toolchainType().toolchainType());
 
     // Find the right one.
-    return resolveConstraints(
+    ArrayList<String> resolutionTrace = debug ? new ArrayList<>() : null;
```

Comment:
We prefer to use interfaces for types where possible, can you change this to:
```
List<String> = debug ? new ArrayList<>() : null;
```

Reply:
Yes sure, will do it.